examples/honest.rs

```diff
@@ -0,0 +1,323 @@
+            }
+        }
+    } else {
+        panic!("{}", USAGE_MSG);
+    }
+}
+
+#[tokio::main]
+async fn main() {
+    env_logger::builder()
+        .filter_module("honest", log::LevelFilter::Info)
+        .init();
+
+    let my_id = parse_arg(1);
+    let n_members = parse_arg(2);
+    let n_finalized = parse_arg(3);
+
+    info!("Getting network up.");
+    let (network, mut manager) = Network::new().await.unwrap();
+    let (close_network, exit) = oneshot::channel();
+    tokio::spawn(async move { manager.run(exit).await });
+
+    let (data_io, mut to_finalize) = DataIO::new();
+
+    let mut interval = tokio::time::interval(Duration::from_millis(500));
+    interval.tick().await;
+
+    while network.connected_peers() != n_members - 1 {
+        info!(
+            "waiting for peers, missing {}",
+            n_members - network.connected_peers() - 1
+        );
+        interval.tick().await;
+    }
+
+    let (close_member, exit) = oneshot::channel();
+    tokio::spawn(async move {
+        let keybox = KeyBox {
+            index: my_id.into(),
+        };
+        let config = rush::default_config(n_members.into(), my_id.into(), 0);
+        let member = rush::Member::new(data_io, &keybox, config);
+
+        member.run_session(network, Spawner {}, exit).await
+    });
+
+    let mut finalized = HashSet::new();
+    while let Some(batch) = to_finalize.next().await {
+        for data in batch {
+            if !finalized.contains(&data) {
+                finalized.insert(data);
+            }
+        }
+        info!("Got new batch. Finalized = {:?}", finalized.len());
+        if finalized.len() == n_finalized {
+            break;
+        }
+    }
+    close_member.send(()).expect("should send");
+    close_network.send(()).expect("should send");
+}
+
+#[derive(PartialEq, Eq, Clone, Debug)]
+struct Hasher64;
+
+impl rush::Hasher for Hasher64 {
+    type Hash = [u8; 8];
+
+    fn hash(x: &[u8]) -> Self::Hash {
+        let mut hasher = DefaultHasher::new();
+        hasher.write(x);
+        hasher.finish().to_ne_bytes()
+    }
+}
+
+type Data = u64;
+struct DataIO {
+    next_data: Arc<Mutex<u64>>,
+    finalized_tx: UnboundedSender<OrderedBatch<Data>>,
+}
+
+impl rush::DataIO<Data> for DataIO {
+    type Error = ();
+    fn get_data(&self) -> Data {
+        let mut data = self.next_data.lock();
+        *data += 1;
+
+        *data
+    }
+    fn send_ordered_batch(&mut self, data: OrderedBatch<Data>) -> Result<(), Self::Error> {
+        self.finalized_tx.unbounded_send(data).map_err(|_| ())
+    }
+}
+
+impl DataIO {
+    fn new() -> (Self, UnboundedReceiver<OrderedBatch<Data>>) {
+        let (finalized_tx, finalized_rx) = mpsc::unbounded();
+        (
+            DataIO {
+                next_data: Arc::new(Mutex::new(0)),
+                finalized_tx,
+            },
+            finalized_rx,
+        )
+    }
+}
+
+#[derive(Clone, Debug, Encode, Decode)]
+struct Signature;
```

Comment:
This is _definitely_ not cryptographically secure. :D

It's fine to have these, but I think it's a good idea to point out in comments what these types _actually_ should satisfy.

Reply:
I wasn't suppose to be, adding comment.